packages/server/src/login/login.controller.ts

```diff
@@ -0,0 +1,140 @@
+import {
+  Controller,
+  Get,
+  Res,
+  Query,
+  Post,
+  Body,
+  UseGuards,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Connection } from 'typeorm';
+import { Response } from 'express';
+import { JwtService } from '@nestjs/jwt';
+import { UserModel } from '../../src/profile/user.entity';
+import { CourseModel } from '../../src/course/course.entity';
+import { UserCourseModel } from '../../src/profile/user-course.entity';
+import {
+  Role,
+  KhouryRedirectResponse,
+  KhouryDataParams,
+} from '@template/common';
+import { NonProductionGuard } from '../../src/non-production.guard';
+import { ConfigService } from '@nestjs/config';
+
+@Controller()
+export class LoginController {
+  constructor(
+    private connection: Connection,
+    private jwtService: JwtService,
+    private configService: ConfigService,
+  ) {}
+
+  @Post('/khoury_login')
+  async recieveDataFromKhoury(
+    @Body() body: KhouryDataParams,
+  ): Promise<KhouryRedirectResponse> {
+    let user: UserModel;
+    user = await UserModel.findOne({
+      where: { username: body.username },
+      relations: ['courses'],
+    });
+
+    if (!user) {
+      user = await UserModel.create({ courses: [] });
+    }
+
+    user = Object.assign(user, {
+      username: body.username,
+      email: body.email,
+      name: body.first_name + body.last_name,
+      photoURL: body.photo_url, // We'll have to find away around this becuase the photo urls expire
+    });
+    await user.save();
+
+    async function courseNameToUserCourse(
```

Comment:
this is an N+1. If someone is in 5 courses, this will endpoint will make ~25 database queries. I am fine leaving it in for now, but could you add a TODO? I am looking into application performance monitoring solutions so we can track the performance of each endpoint

Reply:
Good call. Added that TODO

Reply:
Good call. I added that TODO